src/decision_model.jl

```diff
@@ -140,15 +155,20 @@ function PathCompatibilityVariables(model::Model,
     N = length(diagram.S)
     variables_x_s = Dict{Path{N}, VariableRef}(
         s => path_compatibility_variable(model, (names ? "$(name)$(s)" : ""))
-        for s in paths(diagram.S, fixed)
+        for s in paths(get_values(diagram.S), fixed)
         if !iszero(diagram.P(s)) && !is_forbidden(s, forbidden_paths)
     )
 
     x_s = PathCompatibilityVariables{N}(variables_x_s)
 
     # Add decision strategy constraints for each decision node
-    for (d, z_d) in zip(z.D, z.z)
-        decision_strategy_constraint(model, diagram.S, d, diagram.I_j[d], z.D, z_d, x_s)
+    I_j_indexed = Vector{Node}.([indices_of(diagram, nodes) for nodes in get_values(diagram.I_j)])
+    z_keys_indexed = Node.([index_of(diagram, node) for node in get_keys(diagram.D)])
+
+    z_z = [decision_node.z for decision_node in get_values(z)]
```

Comment:
I think we could get rid of most of these index_of searches by simply adding a running ordering number to the node struct. Then we could just reduce the dicts to the order numbers when we need the node location in the path.

Reply:
Would the new node index trick allow us to write this in a more compact form?